[FLINK-5004] [runtime] Add option to disable queryable state

uce opened the pull request:
- By default, the queryable state server is enabled (current behaviour)
- Via config option `query.server.start` it can be disabled
- In tests, we disable it by default (see mini cluster change)
- Migrated config to `QueryableStateOptions`; not part of the TaskManagerOptions as there are also client options available

StephanEwen commented on flink-tests/src/test/java/org/apache/flink/test/query/QueryableStateITCase.java:
```diff
@@ -115,8 +115,9 @@ public static void setup() {
 			config.setInteger(ConfigConstants.TASK_MANAGER_NUM_TASK_SLOTS, NUM_SLOTS_PER_TM);
-			config.setInteger(ConfigConstants.QUERYABLE_STATE_CLIENT_NETWORK_THREADS, 1);
-			config.setInteger(ConfigConstants.QUERYABLE_STATE_SERVER_NETWORK_THREADS, 1);
+			config.setInteger(QueryableStateOptions.CLIENT_NETWORK_THREADS.key(), 1);
```
You can drop the `key()` part and directly use the option.

StephanEwen commented on flink-core/src/main/java/org/apache/flink/configuration/QueryableStateOptions.java:
```diff
@@ -0,0 +1,72 @@
+
+	/** Flag to indicate whether to start the queryable state server. */
+	public static final ConfigOption<Boolean> START_SERVER =
+			key("query.server.start").defaultValue(true);
```
I other places, we put the default value on the next line. Assuming more aspects to come (deprecated keys, documentation text), this makes it more readable.

uce commented:
Thanks for review. Renamed it to `enable`, removed the `key` calls and added the line breaks. I also removed the `key` calls in other places where they were not needed.
